src/components/Editable.tsx

```diff
@@ -555,9 +595,32 @@ const Editable = ({
         e.preventDefault()
       }
     },
-    [contentRef, editingOrOnCursor, fontSize, allowDefaultSelection, hasMulticursor],
+    [contentRef, editingOrOnCursor, fontSize, allowDefaultSelection, hasMulticursor, handleVoidAreaTap],
   )
 
+  // Manually attach touchstart listener with { passive: false } to allow preventDefault
+  useEffect(() => {
+    const editable = contentRef.current
+    if (!editable) return
+
+    /** Handle touch events to prevent initial cursor jump (Safari-safe). */
+    const handleTouchStart = (e: TouchEvent) => {
+      if (editingOrOnCursor && !hasMulticursor && e.touches.length > 0) {
+        // If not a void area tap, allow browser's default selection
+        if (!handleVoidAreaTap(e.touches[0].clientX, e.touches[0].clientY, () => e.preventDefault())) {
+          allowDefaultSelection()
+        }
+      }
+    }
+
+    // Add event listener with { passive: false } to allow preventDefault
+    editable.addEventListener('touchstart', handleTouchStart, { passive: false })
```

Comment:
Can you clarify a little more how this manual event handling differs from using React `onTouchStart`?

Reply:
Manual event handling is necessary here because react doesn't specifiy {passive: false} when attaching the listener.  https://stackoverflow.com/questions/42101723/unable-to-preventdefault-inside-passive-event-listener, as discussed here, a passive listener cannot call `preventDefault()`. Upon doing so it logs a warning and has no effect. And hence in ios/safari touch devices, default behavior is not prevented and the caret still jumps to the end of the text when clicked in the void area.